core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -152,7 +153,7 @@ abstract class RDD[T: ClassTag](
 
   /** Assign a name to this RDD */
   def setName(_name: String): this.type = {
-    name = _name
+    name = StringUtils.abbreviate(_name, conf.get(MAX_RDD_NAME_LENGTH))
```

Comment:
Hm .. shouldn't we just abbreviate in UI instead of abbreviate the name itself?

Reply:
_name seems only to be used by `RDD.toString` and the `toDebugString` does not depend on it

https://github.com/apache/spark/blob/35848385ae6518b4b72c2f5c1e9ca5a83a190723/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L2008-L2009

Guess it's safe to change? otherwise, we have to try to keep the class name and call site if we abbreviate in UI